src/DiagonalHessianApproximation.jl

```diff
@@ -0,0 +1,217 @@
+using LinearAlgebra
+using LinearOperators
+
+############ function to define products
```

Comment:
Please use a docstring instead of comments. A dosctring is also visible to the user.